src/python/genny/genny_auto_tasks.py

```diff
@@ -0,0 +1,125 @@
+
+    if out.decode() == '':
+        return []
+
+    # Make paths relative to workloads/ e.g. ../workloads/scale/NewTask.yml --> scale/NewTask.yml
+    short_filenames = [f.split('workloads/', 1)[1] for f in out.decode().strip().split('\n')]
+    short_filenames = list(filter(lambda x: x.endswith('.yml'), short_filenames))
+    return short_filenames
+
+
+def get_project_root():
+    """
+    :return: the path of the project root.
+    """
+    try:
+        out = subprocess.check_output('git rev-parse --show-toplevel', shell=True)
+    except subprocess.CalledProcessError as e:
+        sys.exit(e.output)
+
+    return out.decode().strip()
+
+
+def construct_task_json(workloads, variants):
+    """
+    :param list workloads: a list of filenames of workloads to generate tasks for, each in the format subdirectory/Task.yml
+    :param list variants: a list of buildvariants (strings) that the generated tasks should be run on.
+    :return: json representation of tasks that run the given workloads, that can be provided to evergreen's generate.tasks command.
+    """
+    task_specs = []
+    c = Configuration()
+
+    for fname in workloads:
+        basename = os.path.basename(fname)
+        base_parts = os.path.splitext(basename)
+        if base_parts[1] != '.yml':
+            # Not a .yml workload file, ignore it.
+            continue
+
+        task_name = to_snake_case(base_parts[0])
+        task_specs.append(TaskSpec(task_name))
+        t = c.task(task_name)
+        t.priority(5)  # The default priority in system_perf.yml
+        t.commands([
+            CommandDefinition().function('prepare environment').vars({
+                'test': task_name,
+                'auto_workload_path': fname
+            }),
+            CommandDefinition().function('deploy cluster'),
+            CommandDefinition().function('run test'),
+            CommandDefinition().function('analyze'),
+        ])
+
+    for v in variants:
+        c.variant(v).tasks(task_specs)
+
+    return c.to_json()
+
+
+def main():
+    """
+    Main Function: parses args, writes to file evergreen tasks in json format for workloads that have been modified locally.
+    """
+    parser = argparse.ArgumentParser(
+        description="Generates json that can be used as input to evergreen's generate.tasks, representing genny workloads to be run")
+
+    parser.add_argument('--variants', nargs='+', required=True, help='buildvariants that workloads should run on')
+    parser.add_argument('-o', '--output', default='build/generated_tasks.json')
+    args = parser.parse_args(sys.argv[1:])
+
+    workloads = modified_workload_files()
+    if len(workloads) == 0:
+        sys.exit(
+            'No modified workloads found, generating no tasks.\n\
+            No results from command: git diff --name-only --diff-filter=AMR $(git merge-base HEAD origin/master) -- ../workloads/\n\
+            Ensure that any added/modified workloads have been committed locally.')
```

Comment:
let's `print` this and then `return` rather than exiting with it

Reply:
Done.